### PR TITLE
Add support for pd form selects with dynamic options

### PR DESCRIPTION
**Pull request by aoby**
Also modified behavior so the placeholder option disappears once something is selected.

### CODE REVIEW

File: apps/src/code-studio/pd/form_components/FormComponent.jsx [hunk 1]
```diff
@@ -56,17 +56,34 @@ export default class FormComponent extends React.Component {
...
-      options = this.props.options[name].map(value => (
+    const options = this.props.options[name];
+    return this.buildSelectFieldGroup({name, label, placeholder, required, options, props});
+  }
+
+  /**
+   * Construct a controlled Select dropdown with supplied options
+   *
+   * @param {String} name - the name of the input. Should match a key in
+   *        this.props.options
```
Comment (Hamms) [inline apps/src/code-studio/pd/form_components/FormComponent.jsx]: nit: should match a key in options

File: apps/src/code-studio/pd/form_components/FormComponent.jsx [hunk 2]
```diff
@@ -80,9 +97,10 @@ export default class FormComponent extends React.Component {
         onChange={this.handleChange}
         value={this.props.data[name] || ''}
         required={required}
+        {...props}
       >
-        {placeholder && <option key="placeholder">{placeholder}</option>}
-        {options}
+        {placeholder && !this.props.data[name] && <option key="placeholder">{placeholder}</option>}
```
Comment (Hamms) [inline apps/src/code-studio/pd/form_components/FormComponent.jsx]: I think we still want the placeholder if they've already answered, so they can clear their answer. I suppose we _could_ do this if `required === false`, but only then.

File: apps/src/code-studio/pd/form_components/FormComponent.jsx (hunk 1 above)
Comment (aoby) [reply apps/src/code-studio/pd/form_components/FormComponent.jsx]: D'oh of course

File: apps/src/code-studio/pd/form_components/FormComponent.jsx (hunk 2 above)
Comment (aoby) [reply apps/src/code-studio/pd/form_components/FormComponent.jsx]: ok, that makes sense. I'll remove this part

**Comment by aoby** [comment]
Circle failures are unrelated. Merging...

**Pull request merged**